### PR TITLE
Beampipe modifications

### PR DESCRIPTION
**Pull request by Voutsi**
BEGINRELEASENOTES
  * Adding variable "nocore" for beam pipe (with default value = false) In case the variable appears in the BP xml file as "true", the BP sections will have no core of beam material, in order that someone might add various BP walls made of different materials while avoiding G4 overlaps. Example of use:
```xml
<detector name="BeBeampipe" type="DD4hep_Beampipe_o1_v01" insideTrackingVolume="true" nocore="true" vis="BeamPipeVis">
```

ENDRELEASENOTES

### CODE REVIEW

File: DDDetectors/src/Beampipe_o1_v01_geo.cpp [hunk 1]
```diff
@@ -266,9 +262,25 @@ static dd4hep::Ref_t create_element(dd4hep::Detector& description,
...
-	tubeLog2.placeVolume( wallLog2,  Transform3D() );
+	if (nocore){
+
+	  //placement of the wall only
+	  envelope.placeVolume( wallLog,  transformer );
+	  envelope.placeVolume( wallLog2,  transmirror );
+
+	}
+
+	else{
```
Comment (andresailer) [inline DDDetectors/src/Beampipe_o1_v01_geo.cpp]: Please put the else on the line with the closing bracket

**Comment by andresailer** [comment]
Can you post or link to an example of how the beampipe looks with `nocore=true`? Thanks

File: DDDetectors/src/Beampipe_o1_v01_geo.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ static dd4hep::Ref_t create_element(dd4hep::Detector& description,
   //Access to the XML File
   xml_det_t xmlBeampipe = element;
   const std::string name = xmlBeampipe.nameStr();
-
+  bool nocore  = xmlBeampipe.nocore(false);
```
Comment (Voutsi) [inline DDDetectors/src/Beampipe_o1_v01_geo.cpp]: Yes, the idea is to put several BP detectors, the inner one (e.g. coating) with nocore=false and then several ones around it w/o core, like Be, water cooling tube, shields etc... I think that would be easier if I put it as a detector attribute, at least for the FCCee case. Your idea is to add an additional section attribute or to replace the detector attribute with a section one?

**Comment by Voutsi** [comment]
You can find an example of an xml file with nocore = true at:
/afs/cern.ch/user/v/voutsina/public/Beampipe_o4_v02_test_new.xml

File: DDDetectors/src/Beampipe_o1_v01_geo.cpp (hunk 2 above)
Comment (andresailer) [inline DDDetectors/src/Beampipe_o1_v01_geo.cpp]: Ok, I see. I hadn't thought about placing two beampipe detectors. I think then it is OK for me.

**Comment by andresailer** [comment]
Can you now just squash your commits, please? At least getting rid of the merge commits, and the whitespace only commits.

**Comment by andresailer** [comment]
Ok, thanks for cleaning up the commits. (The commit about "rootma[" will be removed by rebasing).
Is there anything else you want to add? Otherwise I will merge.

**Comment by Voutsi** [comment]
Thanks, no I don't have anything else concerning this request

**Pull request merged**